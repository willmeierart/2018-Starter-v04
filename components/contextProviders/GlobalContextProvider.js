import React, { Component } from 'react'

const GlobalContext = React.createContext()

class GlobalContextProvider extends Component {
  constructor (props) {
    super(props)
    this.state = {
      val: 'A'
    }
  }

  changeVal = () => {
    this.setState({ val: this.state.val === 'A' ? 'B' : 'A' })
  }

  render () {
    return (
      <GlobalContext.Provider value={{
        val: this.state.val,
        changeVal: this.changeVal}}>
        { this.props.children }
      </GlobalContext.Provider>
    )
  }
}

export default GlobalContextProvider
