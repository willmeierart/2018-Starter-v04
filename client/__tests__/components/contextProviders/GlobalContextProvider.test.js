/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import GlobalContextProvider from '../../../components/__contextProviders/GlobalContextProvider'

describe('GlobalContextProvider', () => {
  it('renders correctly', () => {
    const globalContextProvider = shallow(<GlobalContextProvider />)
    expect(globalContextProvider).toMatchSnapshot()
  })
})
