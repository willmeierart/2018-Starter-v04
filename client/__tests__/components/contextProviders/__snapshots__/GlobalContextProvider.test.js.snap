// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalContextProvider renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GlobalContextProvider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "value": Object {
        "changeVal": [Function],
        "val": "A",
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "unstable_read": [Function],
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "value": Object {
          "changeVal": [Function],
          "val": "A",
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "unstable_read": [Function],
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
